big_tests/tests/pubsub_tools.erl

```diff
@@ -450,21 +450,30 @@ check_subscription(Subscr, Jid, NodeName, Options) ->
             <<"pending">> = exml_query:attr(Subscr, <<"subscription">>)
     end.
 
-check_items(ReceivedItemsElem, ExpectedItemIds, NodeName, WithPayload) ->
+check_items(ReceivedItemsElem, ExpectedItemIds, NodeName) ->
     NodeName = exml_query:attr(ReceivedItemsElem, <<"node">>),
     ReceivedItems = exml_query:subelements(ReceivedItemsElem, <<"item">>),
-    [check_item(ExpectedItemId, WithPayload, ReceivedItem) ||
+    [check_item(ExpectedItemId, ReceivedItem) ||
         {ReceivedItem, ExpectedItemId} <- lists:zip(ReceivedItems, ExpectedItemIds)].
 
-check_item(ExpectedItem, WithPayload, ReceivedItem) ->
-    #{id := ExpectedItemId} = ExpectedItemMap = decode_expected_item(ExpectedItem),
-    ExpectedItemId = exml_query:attr(ReceivedItem, <<"id">>),
-    Content = item_content(WithPayload),
-    ExpectedContent = maps:get(content, ExpectedItemMap, Content),
-    ExpectedContent = exml_query:subelement(ReceivedItem, <<"entry">>).
+check_item(ExpectedItem, ReceivedItem) ->
+    ExpectedItemMap = decode_expected_item(ExpectedItem),
+    maps:map(fun(K, V) ->
+    KBin = atom_to_binary(K, utf8),
+                     check_item_field(KBin, V, ReceivedItem) end, ExpectedItemMap).
+
+check_item_field(Field = <<"entry">>, ExpectedValue, ReceivedItem) ->
+    ExpectedValue = exml_query:subelement(ReceivedItem, Field);
+check_item_field(Field, ExpectedValue, ReceivedItem) ->
+    ExpectedValue = exml_query:attr(ReceivedItem, Field).
```

Comment:
I'm afraid it will be hard to deduce which attribute didn't match. I think the following
```suggestion
    case exml_query:attr(ReceivedItem, Field) of
        ExpectedValue ->
            ok;
        Value ->
            ct:fail("Expected value for attribute ~s was ~s, received ~s", [Field, ExpectedValue, Value]
    end.
```